### PR TITLE
APP-728: Fix build warnings

### PR DESCRIPTION
**Pull request by hugokallstrom**
This PR fixes about 60-70% of build errors. Whats left is implementing the new onActivityResult API, remove deprecated graphQL queries and some various warnings that takes a bit more refactoring to remove.

### Checklist

- [ ] Functionality is covered by an integration test
- [ ] Functionality is accessible in engineering mode

### CODE REVIEW

File: app/src/main/java/com/hedvig/app/feature/chat/ui/AttachFileAdapter.kt [hunk 1]
```diff
@@ -132,7 +132,7 @@ class AttachFileAdapter(
                 uploadFile: (Uri) -> Unit
             ) {
                 binding.apply {
-                    val image = attachImageData[position - 1]
+                    val image = attachImageData[absoluteAdapterPosition - 1]
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/chat/ui/AttachFileAdapter.kt:135]:
```suggestion
                    val image = attachImageData[bindingAdapterPosition - 1]
```

They are the same in this case, but for the list to line up if it's a sub-adapter, this is needed 😉

File: app/src/main/java/com/hedvig/app/feature/keygear/data/DeviceInformationService.kt [hunk 2]
```diff
@@ -36,13 +37,8 @@ class DeviceInformationService(
...
         }
 
-        val windowManager = context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
-        val displayMetrics = DisplayMetrics()
-
-        windowManager.defaultDisplay.getRealMetrics(displayMetrics)
-
-        val widthInches = displayMetrics.widthPixels / displayMetrics.xdpi
-        val heightInches = displayMetrics.heightPixels / displayMetrics.xdpi
+        val widthInches = (context.getActivity()?.windowWidth ?: 0).toFloat()
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/keygear/data/DeviceInformationService.kt:40]: I think in this case we actually want the `displayMetrics`, as we're not trying to figure out stuff about the window, but instead about the executing device 🙂

File: app/src/main/java/com/hedvig/app/util/extensions/ActivityExtensions.kt [hunk 3]
```diff
@@ -4,31 +4,54 @@ import android.app.Activity
...
 import com.hedvig.app.R
 import com.hedvig.app.authenticate.AuthenticateDialog
 import com.hedvig.app.feature.chat.ui.ChatActivity
 import com.hedvig.app.feature.offer.ui.OfferActivity
 import com.hedvig.app.util.extensions.view.setupToolbar
 import e
 
-val Activity.displayMetrics: DisplayMetrics
+val Activity.screenWidth: Int
+    get() = window.decorView.measuredWidth
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/util/extensions/ActivityExtensions.kt:26]: Is this correct? 🤔

**Review by hedvigoscar — APPROVED** [review]
Marvelous! Check out the comments, and make sure to run `ktlint` 🙁

File: app/src/main/java/com/hedvig/app/util/extensions/ActivityExtensions.kt (hunk 3 above)
Comment (hugokallstrom) [reply app/src/main/java/com/hedvig/app/util/extensions/ActivityExtensions.kt:26]: I haven't changed this (just moved) so I suppose its correct

**Pull request merged**